### PR TITLE
fix: unset column tags before dropping a column

### LINKED ISSUE #1323: Remove governed tags before dropping a column
### Describe the bug

When attempting to drop a column that has a governed tag, the operation fails because dbt-databricks doesn't remove the tags before dropping the column.

### Steps To Reproduce

1. Add a governed tag to a column using a service principal
2. Modify your dbt model to remove that column 
3. Run the model using the service principal that has rights to the governed tag
4. The column drop operation will fail because the column still has a governed tag 

### Expected behavior

Before dropping a column, dbt-databricks should automatically remove any governed tags from the column, allowing the drop operation to succeed while respecting tag governance policies.

### System information

**The output of `dbt --version`:**
```
Core:
  - installed: 1.10.13

Plugins:
  - databricks: 1.10.19
```

**The operating system you're using:**
MacOS

**The output of `python --version`:**
```
Python 3.12.12
```

### PR DESCRIPTION
**Pull request by saishreeeee**
Resolves #1323 

### Description

Dropping a column that carries a governed tag fails with CANNOT_DROP_TAGGED_COLUMN, because Unity Catalog requires the tag be removed before the column. This unsets any tags on the columns being dropped (via on_schema_change='sync_all_columns') before issuing DROP COLUMNS, so tagged columns can be dropped cleanly.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.
- [ ] [Optional] I have run `/dbt-databricks-pr-ready` (AI agent skill in `.claude/skills/`) and addressed its merge-readiness feedback

### CODE REVIEW

File: tests/functional/adapter/column_tags/test_column_tags.py [hunk 1]
```diff
@@ -157,6 +157,55 @@ def test_unchanged_rerun_keeps_column_tags(self, project):
         assert actual_tags == expected_tags
 
 
+@pytest.mark.skip_profile("databricks_cluster")
```
Comment (sd-db) [inline tests/functional/adapter/column_tags/test_column_tags.py:176]: ordinary databricks_tags don’t raise `CANNOT_DROP_TAGGED_COLUMN` — that only happens with governed tags. Suggested add-on test creates a unique governed tag and soft-skips if the identity lacks account CREATE. Gist for change - https://github.com/saishreeeee/dbt-databricks/compare/PECOBLR-3702...sd-db:suggestion/pr-1597-governed-tag-test

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 ### Fixes
 
+- Allow dropping a column that has governed tags ([#PR](https://github.com/databricks/dbt-databricks/pull/1597) resolves [#1323](https://github.com/databricks/dbt-databricks/issues/1323))
```
Comment (sd-db) [inline CHANGELOG.md]: nit: `#PR` --> `#1597`

File: dbt/include/databricks/macros/relations/components/column_tags.sql [hunk 3]
```diff
@@ -48,6 +48,42 @@
   )
 {%- endmacro -%}
 
+{% macro unset_column_tags(relation, columns) -%}
```
Comment (sd-db) [inline dbt/include/databricks/macros/relations/components/column_tags.sql:51]: Can we add some more tests for `unset_column_tags` orchestration (HMS/empty/filter)

**Review by sd-db — APPROVED** [review]
Changes look good !
